app/services/promotional_tweet/post_promotional_tweet.rb

```diff
@@ -0,0 +1,42 @@
+class PostPromotionalTweet
+
+  def initialize(promotional_tweet, image, brand)
+    @message = promotional_tweet.message
+    @image   = image
+    @brand   = brand
+    @client  = brand.twitter_rest_client
+  end
+
+  def send!
+    @tweet = post_promo_tweet
+
+    if @tweet.present?
+      @promotional_tweet.update_attribute!(:tweet_url, @tweet.id)
```

Comment:
Also, where is `@promotional_tweet` defined? You should consider writing specs for this.